rpc/kate-rpc/src/lib.rs

```diff
@@ -141,6 +148,77 @@ where
 	SDFilter: Filter<CallOf<Block>> + 'static,
 	<<Block as BlockT>::Extrinsic as Extrinsic>::Call: Clone,
 {
+	fn query_rows(
+		&self,
+		rows: Vec<u32>,
+		at: Option<HashOf<Block>>,
+	) -> RpcResult<Vec<Option<Vec<u8>>>> {
+		let block_id = self.block_id(at);
+
+		let signed_block = self
+			.client
+			.block(&block_id)
+			.map_err(|e| internal_err!("Invalid block number: {:?}", e))?
+			.ok_or_else(|| internal_err!("Missing block {}", block_id))?;
+		let block_hash = signed_block.block.header().hash();
+
+		let mut block_ext_cache = self
+			.block_ext_cache
+			.write()
+			.map_err(|_| internal_err!("Block cache lock is poisoned .qed"))?;
+
+		let block_length: BlockLength = self
```

Comment:
`block_length` only used inside `if`-block. Pls move it.

Reply:
This affects three RPCs, I didn't wanted to spend much there, but RPCs are almost the same and code should be reused at some point.